File: wun-ios-example/Sources/WunExample/WunExample.swift
// User-namespace components that participate in the same Wun
// registry the framework's `:wun/*` renderers register through.
//
// Host apps wire this in alongside `WunFoundation.register(into:)`:
//
//     let registry = Registry()
//     WunFoundation.register(into: registry)
//     WunExample.register(into: registry)
//
// The component keyword (here `:myapp/Greeting`) is exactly the
// same shape the brief's user-defined `:myapp/RichEditor` example
// uses, mirrored across the cljc `defcomponent` on the server.

import Foundation
import Wun

public enum WunExample {
    public static func register(into registry: Registry) {
        registry.register("myapp/Greeting", Greeting.render)
        // AUTO-REGISTER-MARK
    }

    public static let registeredComponents: [String] = [
        "myapp/Greeting",
    ]
}

File: wun-android-example/src/main/kotlin/myapp/example/WunExample.kt
// Mirror of WunExample in wun-ios-example: a separate Gradle build
// that registers `:myapp/Greeting` into a Wun Registry alongside
// whatever foundational `:wun/*` renderers the host app has wired
// up. Demonstrates the brief's "no privileged path" claim on the
// Compose / Android stack.

package myapp.example

import wun.Registry

object WunExample {
    fun register(registry: Registry) {
        registry.register("myapp/Greeting", GreetingRenderer.render)
    }

    fun registerDefaults() = register(Registry.shared)

    val components: List<String> = listOf("myapp/Greeting")
}
